### ISSUE TITLE
How do I find out what oktaConvertGroupbyId does?

### ISSUE THREAD

**Issue by Wq3VDUAK29DdpZZkQTSnGYFY3E** [mbegan/Okta-PSModule#48]
I am having difficulty identifying what oktaConvertGroupbyId does.  I see the /convert in the API concerning the Group but can't find it in the Okta API documentation.

[string]$resource = '/api/internal/groups/' + $gid + '/convert'
[string]$method = "Post"

What does oktaConvertGroupbyId actually do?  Is it the "Convert Assignments" button in the UI?

**Comment by mbegan** [comment]
i vaguely recall the purpose of it (also it is not intended to be a public API)

In the early days of [Okta group push](https://help.okta.com/en/prod/Content/Topics/Directory/Directory_Using_Group_Push.htm)

there wasn't anything in the UI to convert an application group to a push group.

So it is NOT the "convert assignments" button.

if you wanted to that it would be a matter of updating the [application users](https://developer.okta.com/docs/reference/api/apps/#application-user-model) `scope`

You could use `oktaGetAppProfilebyUserId ` and `oktaUpdateAppProfilebyUserId` to automate the "convert assignment" button.

-Matt

**Comment by Wq3VDUAK29DdpZZkQTSnGYFY3E** [comment]
I am trying to use the oktaUpdateAppProfilebyUserId to update the "scope" or the "profile.scope" from "User" to "Group" but failing.  There is a similar cmdlet, oktaUpdateUserProfilebyID, referenced here: https://github.com/mbegan/Okta-PSModule/issues/11.  I tried using similar syntax but am not having any luck.

I have tried -partial as well as trying to replace the entire profile but cannot get a change to commit.

Here is what I have so far:
`$AllUserScopes=$AllOktaAppProfileData | Where-Object {$_.appname -eq "Test App" -and $_.scope -eq "User"}

foreach ($UserScope in $AllUserScopes){
    $CurrentAppProfile = oktaGetAppProfilebyUserId -oOrg prev -aid $UserScope.appid -uid $UserScope.userid | select * -ExpandProperty profile
    $UpdatedAppProfile = $CurrentAppProfile
    $UpdatedAppProfile.profile.scope = 'GROUP'
    oktaUpdateAppProfilebyUserId -oOrg prev -partial -aid $UserScope.appid -uid $UserScope.userid -profile $UpdatedAppProfile.profile.scope
}`

I can supply errors if you like but I think I just need another set of eyes.
